ui/dashboards/src/components/Variables/VariableEditor.tsx

```diff
@@ -77,6 +79,26 @@ export function VariableEditor(props: {
     });
   };
 
+  const changeVariableOrder = (index: number, direction: 'up' | 'down') => {
+    setVariableDefinitions((draft) => {
+      if (direction === 'up') {
+        if (index === 0) {
+          return;
+        }
+        const tmp = draft[index - 1] as VariableDefinition;
```

Comment:
Not sure how but wonder if there's some way to avoid the `as VariableDefinition` type casting 🤔

Reply:
> Not sure how but wonder if there's some way to avoid the `as VariableDefinition` type casting 🤔

Ah yeah I wrestled with the type system here but I updated it to avoid casting.